### PR TITLE
Split codebase into separate files

### LINKED ISSUE #4: Separation of Concerns
It would greatly benefit current and future contributors if the HTML, CSS, and JavaScript were separated into their own files.

### PR DESCRIPTION
**Pull request by fnky**
This PR splits the codebase into separate files in order to improve readability and maintainability.

- Scripts are defined within `main.js`.
- CSS are defined in `styles.css`
- HTML is declared within `index.html`

This also introduces additional changes:

- Properly declare the HTML5 document.
- Prettified codebase—using [prettier](https://prettier.io/)
- HTML escape implementation from #5

Closes #4

### CODE REVIEW

**Review by AhiVT — CHANGES_REQUESTED** [review]
Great work! This is exactly what this project needed! ~~There are just some 'linting' things that need to be done to bring it to spec. Specifically, I noticed that in your HTML tags you put the `>` character on a new line. This may be confusing to contributors and should be placed on the same line as the declared tag. Eg: `<label` and `</label` should be declared as `<label>` and `</label>` respectively. You can find an example of these linting issues on lines 165 - 167 and 179 - 185 to name a few. I've also noticed that your CSS is improperly formatted causing the site layout to not render correctly. In CSS, rules must be specified on **one line**. Eg. Line 1 in your `styles.css` should read `*, *:before, *:after {`~~
**I stand corrected, please see my next comment**

**Comment by AhiVT** [comment]
I noticed when I ran a local copy of your PR to fix the CSS formatting the site remained broken after I fixed the CSS rule. **I've pinpointed the issue to the change from XHTML to HTML5.**
### Here's why that is:
When browsers like Chrome read a HTML without `<!DOCTYPE html>` or they are written in XHTML the browser will render the page in quirk-mode. Quirks mode changes the fundamental properties of how CSS works with the DOM. Since this site was written using Quirks Mode, the CSS rules worked around these Quirks. A list of CSS differences in Quirk Mode can be found [here](https://developer.mozilla.org/en-US/docs/Mozilla/Mozilla_quirks_mode_behavior)

**TL; DR**: Quirk-mode must be enabled to properly render the site. I've created a PR with this change for you @fnky

**Comment by fnky** [comment]
Hey @Kexerval thanks for the review.

The reason for the `>` on a second line is due to prettier accommodating for sensitive whitespacing in HTML as well as keeping everything within 80 characters as much as possible.

You can read more about how [HTML whitespace sensitivity](https://prettier.io/docs/en/options.html#html-whitespace-sensitivity) works in Prettier.

In order to help future contributors, we can add a pre-commit hook using [lint-staged](https://github.com/okonet/lint-staged) and [husky](https://github.com/typicode/husky) to automatically lint and fix formatting upon commits. This also keeps the code consistent and means that you don't have to think about formatting it yourself, but just let prettier do it for you.

This would require people to have npm installed, though and seeing that this repo isn't currently using npm, we could take steps to convert it to one.

Regarding the quirks mode, I had a look and it seemed that the `#wrapper` selector didn't have a proper height to make everything full-height. This has been fixed.

I don't quite know what you mean by converting to CSS 3.0 standards; I tried to validate the CSS in the [CSS Validation Service](https://jigsaw.w3.org/css-validator) and everything seemed fine.

I also validated the HTML using the [Markup Validation Service](https://validator.w3.org) and after adding the missing `alt` attribute to an image, it also passes.

**Comment by fnky** [comment]
I sorted out the HTML wrapping for the tags by exceeding the line width limit to 120.

**Comment by AhiVT** [comment]
Regarding "Standards CSS," this basically means that no special rulesets (Such as the Quirks Ruleset `layout/style/quirk.css`) will be applied by the browser renderer. I was able to verify that the CSS written for the site relied on a Quirks Ruleset by removing `<!DOCTYPE html>` from line 1 in commit [8a8d402](https://github.com/timeshifter/twitch-filtered-chat/blob/8a8d402b0eb25b73295853ad6917baf3eed3977a/index.html). This rendered the page correctly without any additional CSS changes. Even so, the Quirks Ruleset should only be used for webpages made using older HTML versions. 

But there's great news! You've made the changes needed to resolve #7 in commit [82c440d](https://github.com/timeshifter/twitch-filtered-chat/pull/6/commits/82c440d0569793e25806cdb95955c9532069f392) so we don't have to worry about Quirks Mode any longer.

Thank you for taking the time to explain Prettier!

**Review by AhiVT — APPROVED** [review]
Webpage renders correctly from HEAD!

**Comment by AhiVT** [comment]
@timeshifter Bumping for your review

File: main.js [hunk 1]
```diff
@@ -0,0 +1,616 @@
...
+  }
+
+  message = EscapeHtml(message);
+
+  var message_col = '';
+  if (message.indexOf('ACTION') == 0) {
+    message_col = `color: ${user_col}`;
+    message = message.substring(8, message.length - 1);
+  }
+
```
Comment (Chris-Johnston) [inline main.js:400]: Missing chat message filtering from 985d77439aa0f5c4e8861a05a963024b27ec20c8
```suggestion
  message = message.replace(/</g, '&lt;').replace(/>/g, '&gt');
```

**Review by Chris-Johnston — CHANGES_REQUESTED** [review]
I like the idea of using a pre-commit hook to prettify the CSS. I think it may be helpful to include instructions on how to Prettier manually in the Readme, as well.

File: main.js (hunk 1 above)
Comment (AhiVT) [reply main.js:400]: Does this change reside within the if statement on line 396 or outside of it? Unclear from suggested change overview.

Comment (Chris-Johnston) [reply main.js:400]: Just outside it. Now that you mention it, I need to fix the indentation on that line, so I'll do that now.

**Comment by AhiVT** [comment]
@fnky timeshifter has merged #5 which makes this PR conflict. Any chance to resolve? Thanks!

**Comment by fnky** [comment]
I had to force push changes due to some incompatibilities between commits.

This branch already introduced the same escape function, although only slightly different in terms of naming.

It is now up-to-date with `master`.
